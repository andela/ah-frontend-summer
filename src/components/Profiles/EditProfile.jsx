import React from 'react';
import { 
    Form, 
    Button, 
    Grid, 
    Image, 
    Icon, 
    Segment, 
    Container, 
    GridColumn 
} from "semantic-ui-react";
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import '../../assets/styles/SignupForm.scss';
import '../../assets/styles/Profile.scss';

const EditProfile = props => {
    const { 
        onSubmitHandler, 
        profile, 
        imageHandler, 
        onChangeHandler, 
        firstName, 
        lastName, 
        bio, 
        profileLoading, 
        loadingEditButton, 
        disabled 
    } = props;

    return (
        <Container>
            <Segment padded="very" loading={profileLoading} color='teal'>
                <Grid columns={2}>
                    <GridColumn>
                        <Form onSubmit={onSubmitHandler}>
                            <Container textAlign="left">
                                <div className='spaceComponents'>
                                    <Form.Input 
                                        fluid 
                                        placeholder='First Name' 
                                        value={firstName} 
                                        name="firstName" 
                                        onChange={onChangeHandler}
                                    />
                                    <Form.Input 
                                        fluid 
                                        placeholder='Last Name' 
                                        value={lastName} 
                                        name="lastName" 
                                        onChange={onChangeHandler}
                                    />
                                    <textarea 
                                        placeholder='Bio' 
                                        value={bio} 
                                        name="bio" 
                                        onChange={onChangeHandler}
                                    />
                                    <Button
                                        basic
                                        color='teal'
                                        type='submit'
                                        loading={loadingEditButton}
                                        disabled={disabled}
                                        className="UpdateButton"
                                    >
                                        Edit Profile
                                    </Button>
                                    <Link to={`/${profile.username}`}>
                                        <Button color='teal'>Cancel</Button>
                                    </Link>
                                </div>
                            </Container>
                        </Form>
                    </GridColumn>
                    <GridColumn>
                        <Container textAlign="right">
                            <Image 
                                src={profile.image} 
                                size='small' 
                                avatar 
                                style={{ borderRadius: "50%" }} 
                            />
                            <div className="FileUploaderContainer">
                                <input 
                                    className="FileUploader" 
                                    type='file' 
                                    name="image" 
                                    onChange={imageHandler}
                                />
                                <br />
                                <br />
                                <Icon size="large" name="pencil alternate" />
                            </div>
                        </Container>
                    </GridColumn>
                    
                </Grid>
            </Segment>
        </Container>
    );
};

EditProfile.propTypes = {
    onSubmitHandler: PropTypes.func.isRequired,
    profile: PropTypes.object.isRequired,
    imageHandler: PropTypes.func.isRequired,
    onChangeHandler: PropTypes.func.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    bio: PropTypes.string.isRequired,
    profileLoading: PropTypes.bool.isRequired,
    loadingEditButton: PropTypes.bool.isRequired,
    disabled: PropTypes.bool.isRequired
};


export default EditProfile;
