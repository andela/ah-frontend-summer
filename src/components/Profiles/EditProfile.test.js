import React from 'react';
import { shallow } from 'enzyme';
import EditProfile from './EditProfile';

describe("EditProfile", () => {
    const editProfileProps = { 
        onSubmitHandler: jest.fn(), 
        profile: {username: "", image: ""}, 
        imageHandler: jest.fn(), 
        onChangeHandler: jest.fn(), 
        firstName: "", 
        lastName: "", 
        bio: "", 
        profileLoading: false, 
        loadingEditButton: false,
        disabled: true
    };
    const editProfile = shallow(<EditProfile {...editProfileProps}/>);
    
    it("renders correctly", () => {
        expect(editProfile).toMatchSnapshot();
    });
});
