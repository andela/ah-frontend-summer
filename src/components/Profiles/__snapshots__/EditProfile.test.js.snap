// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile renders correctly 1`] = `
<Container>
  <Segment
    color="teal"
    loading={false}
    padded="very"
  >
    <Grid
      columns={2}
    >
      <GridColumn>
        <Form
          as="form"
          onSubmit={[MockFunction]}
        >
          <Container
            textAlign="left"
          >
            <div
              className="spaceComponents"
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                name="firstName"
                onChange={[MockFunction]}
                placeholder="First Name"
                value=""
              />
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                name="lastName"
                onChange={[MockFunction]}
                placeholder="Last Name"
                value=""
              />
              <textarea
                name="bio"
                onChange={[MockFunction]}
                placeholder="Bio"
                value=""
              />
              <Button
                as="button"
                basic={true}
                className="UpdateButton"
                color="teal"
                disabled={true}
                loading={false}
                type="submit"
              >
                Edit Profile
              </Button>
              <Link
                to="/"
              >
                <Button
                  as="button"
                  color="teal"
                >
                  Cancel
                </Button>
              </Link>
            </div>
          </Container>
        </Form>
      </GridColumn>
      <GridColumn>
        <Container
          textAlign="right"
        >
          <Image
            as="img"
            avatar={true}
            size="small"
            src=""
            style={
              Object {
                "borderRadius": "50%",
              }
            }
            ui={true}
          />
          <div
            className="FileUploaderContainer"
          >
            <input
              className="FileUploader"
              name="image"
              onChange={[MockFunction]}
              type="file"
            />
            <br />
            <br />
            <Icon
              as="i"
              name="pencil alternate"
              size="large"
            />
          </div>
        </Container>
      </GridColumn>
    </Grid>
  </Segment>
</Container>
`;
